crates/recent_projects/src/dev_servers.rs

```diff
@@ -254,45 +265,112 @@ impl DevServerProjects {
         }));
     }
 
-    pub fn create_dev_server(&mut self, cx: &mut ViewContext<Self>) {
+    pub fn create_dev_server(&mut self, manual_setup: bool, cx: &mut ViewContext<Self>) {
         let name = get_text(&self.dev_server_name_input, cx);
         if name.is_empty() {
             return;
         }
 
-        let ssh_connection_string = if self.use_server_name_in_ssh == Selection::Selected {
-            Some(name.clone())
-        } else {
+        let ssh_connection_string = if manual_setup {
             None
+        } else {
+            Some(name.clone())
         };
 
         let dev_server = self.dev_server_store.update(cx, |store, cx| {
-            store.create_dev_server(name, ssh_connection_string, cx)
+            store.create_dev_server(name, ssh_connection_string.clone(), cx)
         });
 
+        let workspace = self.workspace.clone();
+
         cx.spawn(|this, mut cx| async move {
             let result = dev_server.await;
 
-            this.update(&mut cx, |this, cx| match &result {
+            match result {
                 Ok(dev_server) => {
-                    this.focus_handle.focus(cx);
-                    this.mode = Mode::CreateDevServer(CreateDevServer {
-                        creating: false,
-                        dev_server: Some(dev_server.clone()),
-                    });
-                }
-                Err(_) => {
-                    this.mode = Mode::CreateDevServer(Default::default());
+                    if let Some(ssh_connection_string) =  ssh_connection_string {
+                    if let Some(terminal_panel) = workspace
+                        .update(&mut cx, |workspace, cx| workspace.panel::<TerminalPanel>(cx))
+                        .ok()
+                        .flatten()
+                        .with_context(|| anyhow::anyhow!("No terminal panel"))
+                        .log_err()
+                    {
+                        let install_command = if env::var("ZED_RPC_URL") == Ok("http://localhost:8080/rpc".to_string()) {
+                            format!(r#"cd; (curl -sSL https://zed.dev/install.sh || wget -qO- https://zed.dev/install.sh) | bash && ZED_RPC_URL=http://localhost:8080/rpc ~/.local/bin/zed --dev-server-token {} && exec $SHELL"#, dev_server.access_token)
+                        } else {
+                            format!(r#"cd; (curl -sSL https://zed.dev/install.sh || wget -qO- https://zed.dev/install.sh) | bash && ~/.local/bin/zed --dev-server-token {}"#, dev_server.access_token)
```

Comment:
```suggestion
                            format!(r#"cd; env zed || (curl -sSL https://zed.dev/install.sh || wget -qO- https://zed.dev/install.sh) | bash && env PATH=$PATH:~/.local/bin zed --dev-server-token {}"#, dev_server.access_token)
```

* only installs zed if it isn't in PATH
* runs zed from PATH (with `~/.local/bin` appended to it, just in case)

Idk if that's a behaviour you want just yet, but it should help towards making it Just Work regardless of how Zed was installed.

Reply:
:+1: I would like something like this, but one tester has already run into https://zed.brimdata.io/docs/commands being already in $PATH, so I only want to use the builtin zed if it's actually the right zed. Seemed easiest for now to just always dump the daemon, but we could do something like parse the output of zed --version.

Reply:
Hmm, right.... Fun times. And the other zed may even be installed to `~/.local/bin`...

I wonder if there's a way to query the location of an application based on its application id (as listed in the .desktop file)

Reply:
Not sure. For now I'm checking for `~/.local/bin/zed` so we at least don't reinstall every time you reconnect (and I pass `-v` which we give exit status 0 for, but the other zed errors).